lib/api/CDataFrameAnalysisInstrumentation.cc

```diff
@@ -512,12 +512,12 @@ void CDataFrameTrainBoostedTreeInstrumentation::writeValidationLoss(json::object
     auto* writer = this->writer();
     if (writer != nullptr) {
         writer->addMember(VALIDATION_LOSS_TYPE_TAG, json::value(m_LossType), parentObject);
-        json::array lossValuesArray{writer->makeArray()};
+        json::array lossValuesArray = writer->makeArray();
```

Comment:
If the braces notation here compiled but produced incorrect results then it's worth adding a comment to say why. Otherwise someone might change it back in the future without realising the subtlety.

Same in the other 3 places.